frontend/src/components/suggestions.tsx

```diff
@@ -7,13 +7,58 @@ export interface SuggestionsProps {
 }
 
 export const Suggestions = ({ loadPrompt, suggestions }: SuggestionsProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isScrollable, setIsScrollable] = useState(false);
+
+  const checkScrollable = useCallback(() => {
+    const container = containerRef.current;
+    if (container) {
+      const computedStyle = window.getComputedStyle(container);
+      const paddingRight = parseFloat(computedStyle.paddingRight) || 0;
+
+      const hasOverflow = container.scrollWidth > container.offsetWidth;
+      const isAtEnd =
+        container.scrollLeft + container.offsetWidth >=
+        container.scrollWidth - paddingRight;
+
+      setIsScrollable(hasOverflow && !isAtEnd);
+    }
+  }, []);
+
+  useEffect(() => {
+    checkScrollable();
```

Comment:
Minor: This is probably not needed as also called for the useEffect below? But will only affect first render.